server/src/main/java/com/objectcomputing/checkins/services/memberprofile/currentuser/CurrentUserServices.java

```diff
@@ -0,0 +1,10 @@
+package com.objectcomputing.checkins.services.memberprofile.currentuser;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+
+import java.util.Map;
+
+public interface CurrentUserServices {
+
+    MemberProfile findOrSaveUser(Map<String, Object> tokenAttributes);
```

Comment:
I'm not a fan of sending this Map directly. It creates an odd coupling of the implementation to the construction of those attributes. I think you would be better off explicitly passing in `workEmail` and `name`.  Let the controller determine how to get that information.